start post about dev guide 0.4.0 (#614)
ropensci/roweb2: content/blog/2020-04-14-devguide-release/index.md

### Policy and governance changes

Some aspects of the software review process changed.

#### Policy changes

We've added field and laboratory reproducibility tools as a category in [scope](https://devguide.ropensci.org/policies.html#aims-and-scope). 
_Packages that improve reproducibility of real-world workflows through standardization and automation of field and lab protocols, such as sample tracking and tagging, form and data sheet generation, interfacing with laboratory equipment or information systems, and executing experimental designs. (Example: [baRcodeR](https://docs.ropensci.org/baRcodeR/))_

{{< tweet 1183811515067043840 >}}

Our [reviewer guide](https://devguide.ropensci.org/reviewerguide.html) now includes guidance for off-thread interactions. We're glad for community building but don't want any important information to get lost.
_If you interact with the package authors and talked about the review outside a review thread (in chats, DMs, in-person, issues in the project repository), please make sure that your review captures and/or links to elements from these conversations that are relevant to the process._

> Another change linked to policy is that we clarified our advice related to rOpenSci's Code of Conduct.
> That code of conduct applies to rOpenSci packages so repositories shouldn't use another Code of Conduct, and the [default `CODE_OF_CONDUCT.md` of the ropensci GitHub organization](https://github.com/ropensci/.github/blob/master/CODE_OF_CONDUCT.md) will be picked up by GitHub in e.g. new issues.

Reviewer: I'm a little confused by what is meant here. If I understand correctly, the rOpenSci code of conduct applies automatically to all repositories so maintainers shouldn't include one? This way it's standardized and will pop up when users open issues etc. I also think that ropensci can be written as rOpenSci because that's the formatting of the organization name on it's page. 

Also the link to the CODE_OF_CONDUCT.md, isn't actually a markdown file with the code of conduct... should it be?

What do you think of the following suggestion?

```suggestion
Another change linked to policy is that we clarified our advice related to rOpenSci's Code of Conduct.
This Code of Conduct applies automatically to all rOpenSci repositories so maintainers should not include a Code of Conduct to avoid conflict. This way the [default `CODE_OF_CONDUCT.md` of the rOpenSci GitHub organization](https://github.com/ropensci/.github/blob/master/CODE_OF_CONDUCT.md) will be picked up by GitHub in e.g. new issues.
```

We recommend authors to add the following text to the package README:

```markdown
Please note that this package is released with a [Contributor Code of Conduct](https://ropensci.org/code-of-conduct/). 
By contributing to this project, you agree to abide by its terms.
```

#### New ethics guidance

Our polices now feature [our expectations regarding "Ethics, Data Privacy and Human Subjects Research"](https://devguide.ropensci.org/policies.html#ethics-data-privacy-and-human-subjects-research).

#### Submission form amendments

Our [issue template for submissions](https://github.com/ropensci/software-review/blob/master/.github/ISSUE_TEMPLATE/A-submit-software-for-review.md) now:

* explicitly mentions [roxygen2](https://roxygen2.r-lib.org/). Prospective submitters, note that if you used to write documentation the hard way, you can convert it using [rd2roxygen]( https://yihui.org/rd2roxygen/).

* explicitly mentions the packaging guide and guide for authors

> * asks whether the submitted package is intented to go on Bionconductor, similarly to what's asked for CRAN.

Reviewer: ```suggestion
* asks whether the submitted package is intended to go on Bioconductor, similar to what's asked for CRAN.
```

If you want to know more about GitHub Actions for R, we recommend [Jim Hester's RStudio conf talk](https://resources.rstudio.com/rstudio-conf-2020/azure-pipelines-and-github-actions-jim-hester) and [Emil Hvitfeldt's thorough walkthrough "Deploy your bookdown project to Netlify with Github Actions"](https://www.hvitfeldt.me/blog/bookdown-netlify-github-actions/). You can also explore [our (probably improvable) workflows](https://github.com/ropensci/dev_guide/tree/dev/.github/workflows).
 
#### URL checking

Our [script checking URLs](https://github.com/ropensci/dev_guide/blob/master/inst/book_grooming.R) now uses [commonmark](/technotes/2018/09/05/commonmark/) instead of regular expressions. If you're interesting in programmatic URL cleaning, you can also read [the recent rOpenSci tech note about cleaning a website URLs with R](/technotes/2019/12/19/urls-tidying/).

#### Conclusion

In this post we summarized the changes incorporated into our book ["rOpenSci Packages: Development, Maintenance, and Peer Review"](https://devguide.ropensci.org/) over the last six months. We are grateful for all contributions that made this release possible. 

> Although our release frequency has now slowed down, we are already working on updates for our next version, such as [how to deal with the documentation website documenting the development vs release version of a package](https://github.com/ropensci/dev_guide/issues/200). 

Reviewer: To avoid the "documentation ... documenting" tongue twister :grin: Does the following still capture the same idea?

```suggestion
Although our release frequency has now slowed down, we are already working on updates for our next version, such as [how to permit multiple documentation websites for development vs release versions of a package](https://github.com/ropensci/dev_guide/issues/200). 
```

Check out the [the issue tracker](https://github.com/ropensci/dev_guide/issues/) if you'd like to contribute.

> [^1]: We first explored this in the blog guide bookdown BLOG POST LINK

Reviewer: Need a link?

Author: Yes, thanks, now added!
